lib/compress/zstd_compress.c

```diff
@@ -140,6 +140,9 @@ struct ZSTD_CCtx_s {
     /* Multi-threading */
     U32 nbThreads;
     ZSTDMT_CCtx* mtctx;
+
+    /* adaptive compression */
+    double completion;
```

Comment:
Can you find another method which doesn't involve tracking a new variable inside `CCtx` ?